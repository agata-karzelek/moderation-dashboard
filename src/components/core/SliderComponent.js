import React from "react";
import { useStyles, SlideBar } from "../../style/slideBar";
import { Slider } from "@material-ui/core";

const PrettoSlider = SlideBar(Slider);

export default function SliderComponent({ value, onChange, name }) {
  const classes = useStyles();

  const marks = [
    {
      value: 0,
      label: name === "tisane" ? "None" : 0,
    },
    {
      value: 0.25,
      label: name === "tisane" ? "Extreme" : 0.25,
    },
    {
      value: 0.5,
      label: name === "tisane" ? "High" : 0.5,
    },
    {
      value: 0.75,
      label: name === "tisane" ? "Medium" : 0.75,
    },
    {
      value: 1,
      label: name === "tisane" ? "Low" : 1,
    },
  ];

  function valuetext(values) {
    return `${values}`;
  }

  return (
    <div className={classes.root}>
      <PrettoSlider
        value={value}
        min={0}
        getAriaValueText={valuetext}
        aria-labelledby="discrete-slider"
        step={0.25}
        marks={marks}
        max={1}
        onChange={onChange}
      />
    </div>
  );
}
