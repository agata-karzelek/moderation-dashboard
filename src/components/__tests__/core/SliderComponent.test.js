import React from "react";
import { shallow } from "enzyme";
import SliderComponent from "../../core/SliderComponent";

describe("Test Cases for Slider Component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <SliderComponent value={0.75} onChange={jest.fn()} name="sightengineRiskFactorThreshold" />
    );
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
