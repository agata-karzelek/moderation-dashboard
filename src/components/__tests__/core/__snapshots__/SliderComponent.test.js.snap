// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cases for Slider Component Snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(WithStyles(ForwardRef(Slider)))
    aria-labelledby="discrete-slider"
    getAriaValueText={[Function]}
    marks={
      Array [
        Object {
          "label": 0,
          "value": 0,
        },
        Object {
          "label": 0.25,
          "value": 0.25,
        },
        Object {
          "label": 0.5,
          "value": 0.5,
        },
        Object {
          "label": 0.75,
          "value": 0.75,
        },
        Object {
          "label": 1,
          "value": 1,
        },
      ]
    }
    max={1}
    min={0}
    onChange={[MockFunction]}
    step={0.25}
    value={0.75}
  />
</div>
`;
